pontoon/sync/formats/compare_locales.py

```diff
@@ -39,6 +43,16 @@ def __init__(self, path, source_resource=None):
         self.source_resource = source_resource
         self.parser = parser.getParser(self.path)
 
+        # Copy entities from the source_resource if it's available.
+        if source_resource:
+            for key, entity in source_resource.entities.items():
+                self.entities[key] = CompareLocalesEntity(
+                    entity.key,
+                    '',
+                    None,
+                    entity.order
+                )
```

Comment:
TODO: Look into refactoring the code so that this block is not required.

Reply:
There are several places where the code expects l10n resources to contain all entities, even if they don't exist in the file, e.g.:
* `pontoon.sync.core.py:update_translations()`
* `pontoon.base.utils.py:get_download_content()`
* `pontoon.base.utils.py:handle_upload_content()`